### ISSUE TITLE
Get SankeyMakie registered

### ISSUE THREAD

**Issue by jkrumbiegel** [jkrumbiegel/SankeyMakie.jl#2]
Looks like this package is actually used by some people here and there, so it would probably be worth getting it registered and having CI set up on it. @simonsteiger I got the impression you were interested in this package, would you be interested in doing the necessary PRs to reach that goal? That would probably just mean setting up some sort of reference tests and cleaning up the API where needed?

**Comment by simonsteiger** [comment]
Oh woops, I totally missed that I was tagged here!

I actually wondered about adding tests when I worked on #6 but skipped it since I was unaware that you're interested in getting it registered.

I've briefly had a look at the tests you wrote for [SwarmMakie.jl](https://github.com/MakieOrg/SwarmMakie.jl) since I wasn't sure how to approach testing a visualisation library. I guess we could test if the nodes end up in the expected `x` coordinates and whether there is any overlap between nodes along the `y` axis?

I'd be happy to hear if you have further ideas for tests.

**Comment by jkrumbiegel** [comment]
The easiest is just using reference tests. They cover a lot of values you could theoretically check in code but at the same time allow you to quickly sanity check your results. You could copy the implementation from AlgebraOfGraphics which is just what Makie does without the overhead of storing images in a separate repo.

**Comment by simonsteiger** [comment]
I recall from MakieOrg/Makie.jl#4304 that we should probably link `lift` to the plot object before registering this package.

So this

https://github.com/jkrumbiegel/SankeyMakie.jl/blob/10bc7676fd714a03951d6f7c8f2b3dc830d58739/src/SankeyMakie.jl#L397

should be

```julia
lift(plt, scene.camera.projectionview, scene.viewport) do _, _
```

... I believe. I can add that in another PR that cleans up the API.

**Comment by jkrumbiegel** [comment]
Yeah, better not to do everything at once. But such a change is not breaking so it can be done whenever.

**Comment by jkrumbiegel** [comment]
So yeah with the CI set up the main thing to do would be some cleanup, docstrings, etc.

**Comment by simonsteiger** [comment]
Sounds good, I'll do that soon!

**Comment by simonsteiger** [comment]
I think setting `linkcolor`s to `TargetColor` or `SourceColor` is a great feature and should be easy to discover for new users. Maybe having a very simple documentation page could take care of that eventually.
